server/src/document/DbtTextDocument.ts

```diff
@@ -201,14 +201,20 @@ export class DbtTextDocument {
   }
 
   async resendDiagnostics(): Promise<void> {
-    if (this.destinationState.contextInitialized && this.dbt.dbtReady && !this.currentDbtError && !this.modelCompiler.compilationInProgress) {
+    if (
+      this.destinationState.contextInitialized &&
+      this.dbt.dbtReady &&
+      !this.currentDbtError &&
+      !this.modelCompiler.compilationInProgress &&
+      this.compiledDocument.getText() !== this.rawDocument.getText()
```

Comment:
If we don't need to resend diagnostics if `this.compiledDocument.getText() !== this.rawDocument.getText()` ? What if model does not contains jinjas?

Reply:
The cause to resend is changes in manifest and in related models. In case model does not contains jinjas it should have fresh diagnostics, so we also don't need to resend.